frontend/src/libs/run.ts

```diff
@@ -25,6 +26,60 @@ export const getStatusIconType = (status: IRun['status']): StatusIndicatorProps[
     }
 };
 
+export const getStatusIconColor = (status: IRun['status'] | TJobStatus, termination_reason: string | null | undefined): StatusIndicatorProps.Color | undefined => {
+    if (termination_reason === 'failed_to_start_due_to_no_capacity') {
+        return 'yellow';
+    }
+
+    switch (status) {
+        case 'aborted':
+            return 'yellow'
+        default:
+            return undefined;
+    }
+};
+
+
+const capitalize = (str: string): string => str.charAt(0).toUpperCase() + str.slice(1);
+
+export const getJobSubmissionStatus = (run: IRun): string => {
```

Comment:
Why duplicating the pretty_repr logic in the frontend/CLI – they'll be likely to diverge. Better to implement it on the server and just show on the CLI/frontend side.